src/PackageUtilities.js

```diff
@@ -201,6 +201,8 @@ export default class PackageUtilities {
         batch.push(packages.reduce((a, b) => {
           return (refCounts[a.name] || 0) > (refCounts[b.name] || 0) ? a : b;
         }));
+        
+        log.silly("packages", batch.reduce((pkgs, pkg) => pkgs.push(pkg.name), []).join(', '));
```

Comment:
[Array.prototype.push()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) actually returns the new length value, not the array pushed into. The `reduce` callback will need to be slightly longer (returning `pkgs` after pushing into it), and I don't think you'll need the `.join(', ')`, as the logger handles displaying arrays across multiple lines fairly well.

Reply:
`log.silly("packages", batch.reduce((pkgs, pkg) => (pkgs.push(pkg.name), pkgs), []))` ^.^

prob something like that could be easier to read:
`log.silly("packages", [].concat.apply([], batch.map(pkg => pkg.name)))`